share/eds/byzantine/bad_encoding_test.go

```diff
@@ -0,0 +1,65 @@
+package byzantine
+
+import (
+	"context"
+	"testing"
+
+	mdutils "github.com/ipfs/go-merkledag/test"
+	"github.com/stretchr/testify/require"
+	core "github.com/tendermint/tendermint/types"
+
+	"github.com/celestiaorg/celestia-app/pkg/da"
+	"github.com/celestiaorg/celestia-node/header"
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/ipld"
+	"github.com/celestiaorg/rsmt2d"
+)
+
+func TestFalsePositiveBadEncodingFraudProof(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	bServ := mdutils.Bserv()
+
+	squareSize := 8
+	shares := share.RandShares(t, squareSize*squareSize)
```

Comment:
@evan-forbes, we have a utiltiy for this, so I just used it. Not sure what was the reasoning behind making share size 504 btw

Reply:
nice! and 504 = 512 - namespaceIDSize